### PR TITLE
[Spike] Re-style Proposal Single page

### PR DESCRIPTION
**Pull request by ken-futureverse**
## Summary

Re-style the Proposal Single page with a 2 columns layout

## Changes

- Add `ProposalBody`, `ProposalSidebar` and `ProposalVoteForm` as the main components to construct the page
- Add `ProposalVoteFormDialog` that follows the current setup convention
- Fix `AccountSelect` doesn't pick up the initial value
- Add `VoteInfo` and `VetoInfo` to `ProposalSidebar` depending on the status
- Use `subscribeFinalizedHeads` to regularly update the proposal details
- Add `votePercentage` to the `Proposal` model

## Notes

Screen Shot 2022-07-20 at 14 26 58

### CODE REVIEW

File: libs/web/components/src/ProposalSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}) => {
+	const { vetoPercentage } = proposal;
+
+	return (
+		<div {...props}>
+			<label className="font-display text-hero block font-bold uppercase">
+				Threshold to Veto
+			</label>
+			<span>
+				Current: {vetoPercentage?.toFixed(2)}% / Require: {vetoThreshold}%
```
Comment (aidan-starke) [inline libs/web/components/src/ProposalSidebar.tsx]:
```suggestion
				Current: {vetoPercentage?.toFixed(2)}% / Required: {vetoThreshold}%
```

File: libs/web/components/src/ProposalVoteFormDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+					<Choose.When condition={formState?.step === "Sign"}>
+						<p className="prose text-center text-sm">
+							Please sign the transaction when prompted...
+						</p>
+					</Choose.When>
+
+					<Choose.When condition={formState?.step === "Submit"}>
+						<p className="prose text-center text-sm">
+							Please wait until this proccess completes...
```
Comment (aidan-starke) [inline libs/web/components/src/ProposalVoteFormDialog.tsx]:
```suggestion
							Please wait until this process completes...
```

File: libs/web/hooks/src/useProposalVoteForm.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+						setFormStatus("Cancelled");
+					},
+				}
+			);
+			setFormState({ step: "Complete", status: "Ok" });
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		} catch (error: any) {
+			console.info(error);
+			setFormStatus("NotOk", `[${error?.code ?? "UNKNOWN"}] ${error?.message}`);
+		}
```
Comment (aidan-starke) [inline libs/web/hooks/src/useProposalVoteForm.ts:87]: Might need to run Prettier format

File: apps/gov-website/pages/proposal/[pid].tsx [hunk 4]
```diff
@@ -41,28 +70,142 @@ export const getStaticProps: GetStaticProps = async (context) => {
...
+	const { proposalId } = initialProposal;
+	const [proposal, setProposal] = useState<ProposalModel>(initialProposal);
+
+	useEffect(() => {
+		if (!api) return;
+		let unsubscribeFn: VoidFn;
+		let lastBlockPolled: number;
+		api.rpc.chain
+			.subscribeFinalizedHeads(async (head) => {
+				const blockNumber = head.number.toNumber();
```
Comment (aidan-starke) [inline apps/gov-website/pages/proposal/[pid].tsx]: could be worth extracting `subscribeFinalizedHeads` to a util that takes a callback?

**Review by aidan-starke — DISMISSED** [review]

File: apps/gov-website/pages/proposal/[pid].tsx (hunk 4 above)
Comment (ken-futureverse) [reply apps/gov-website/pages/proposal/[pid].tsx]: good call

File: libs/web/hooks/src/useProposalVoteForm.ts (hunk 3 above)
Comment (ken-futureverse) [reply libs/web/hooks/src/useProposalVoteForm.ts:87]: hmm nothing changes when I run `yarn format`, how does it show up to you?

File: apps/gov-website/pages/proposal/[pid].tsx (hunk 4 above)
Comment (ken-futureverse) [reply apps/gov-website/pages/proposal/[pid].tsx]: done

File: libs/web/hooks/src/useProposalVoteForm.ts (hunk 3 above)
Comment (aidan-starke) [reply libs/web/hooks/src/useProposalVoteForm.ts:87]: ahh I got tripped up because the one above it `setFormStatus` not `setFormState`

**Review by aidan-starke — APPROVED** [review]

**Pull request merged**